src/syft/core/store/store_memory.py

```diff
@@ -69,16 +69,27 @@ def __getitem__(self, key: UID) -> AbstractStorableObject:
         try:
             return self._objects[key]
         except Exception as e:
-            logger.trace(f"MemoryStore get item error {key} {e}")
+            logger.critical(f"{type(self)} __getitem__ error {key} {e}")
             raise e
 
     @syft_decorator(typechecking=True, prohibit_args=False)
     def __setitem__(self, key: UID, value: AbstractStorableObject) -> None:
         self._objects[key] = value
 
+    @syft_decorator(typechecking=True, prohibit_args=False)
+    def delete(self, key: UID) -> None:
+        try:
+            obj = self.get_object(key=key)
+            if obj is not None:
+                self._objects.__delitem__(key)
+            else:
+                logger.critical(f"{type(self)} __delitem__ error {key}.")
+        except Exception as e:
+            logger.critical(f"{type(self)} Exception in __delitem__ error {key}. {e}")
+
     @syft_decorator(typechecking=True, prohibit_args=False)
     def __delitem__(self, key: UID) -> None:
-        del self._objects[key]
+        self.delete(key=key)
```

Comment:
Q: Can't we delete this method? (```__delitem__```)
I think we should always use:
store.delete(key) <-- and this should delete our object